sklearn/linear_model/least_angle.py

```diff
@@ -916,10 +916,14 @@ def _lars_path_residues(X_train, y_train, X_test, y_test, Gram=None,
     residues : array, shape (n_alphas, n_samples)
         Residues of the prediction on the test data
     """
-    X_train = _check_copy_and_writeable(X_train, copy)
-    y_train = _check_copy_and_writeable(y_train, copy)
-    X_test = _check_copy_and_writeable(X_test, copy)
-    y_test = _check_copy_and_writeable(y_test, copy)
+    X_train = _check_copy_and_writeable(X_train, copy).astype(np.float64,
+                                                              copy=False)
+    y_train = _check_copy_and_writeable(y_train, copy).astype(np.float64,
+                                                              copy=False)
+    X_test = _check_copy_and_writeable(X_test, copy).astype(np.float64,
+                                                            copy=False)
+    y_test = _check_copy_and_writeable(y_test, copy).astype(np.float64,
+                                                            copy=False)
```

Comment:
But the input checks on dtypes should probably be done outside, in the public function or in the fit method.

We do the `_check_copy_and_writeable` check in this function for specific constraints caused by the use of multiprocessing with memmap data.